src/main/resources/public/ts/controllers/controller.ts

```diff
@@ -0,0 +1,788 @@
+				notify.error('support.issue.validation.error.comment.is.empty');
+				$scope.ticket.issue.processing = false;
+				return;
+			}
+
+			const successCallback = function () {
+				$scope.ticket.issue.showEditForm = false;
+				$scope.ticket.issue.processing = false;
+				notify.info('support.comment.issue.successful');
+			};
+
+			const errorCallback = function (error) {
+				notify.error(error);
+				$scope.ticket.issue.processing = false;
+			};
+
+			$scope.ticket.commentIssue(successCallback, errorCallback);
+		};
+
+		// Date functions
+		$scope.formatDate = function(date) {
+			if(/^.*Z.*$/.test(date)){
+				// If Z (=> UTC) is specified, moment will convert to UTC automatically
+				return $scope.formatMoment(moment(date));
+			}
+			// Else, assuming date is in UTC, we convert to UTC by ourselves
+			return $scope.formatMoment(moment.utc(date).local());
+		};
+
+		$scope.formatMoment = function(moment) {
+			return moment.lang('fr').format('DD/MM/YYYY H:mm');
+		};
+
+		// Functions to display proper label
+		$scope.getStatusLabel = function(status) {
+			if(model.ticketStatusEnum.properties[status] !== undefined) {
+				const key = model.ticketStatusEnum.properties[status].i18n;
+				return lang.translate(key);
+			}
+			return undefined;
+		};
+
+		$scope.getCategoryLabel = function(appAddress) {
+			const app = _.find($scope.apps, function (app) {
+				return app.address === appAddress;
+			});
+			return (app !== undefined) ? app.displayName : undefined;
+		};
+
+		$scope.getSchoolName = function(schoolId) {
+			const school = _.find($scope.schools, function (school) {
+				return school.id === schoolId;
+			});
+			return (school !== undefined) ? school.name : undefined;
+		};
+
+		$scope.canEscalate = function(ticket){
+			const canEscalate = model.me.workflow.support.escalate || false;
+			return ($scope.isEscalationActivated === true && canEscalate);
+		};
+
+		$scope.userIsLocalAdmin = function(ticket){
+			// SUPER_ADMIN
+			if( model.me.functions.SUPER_ADMIN ) {
+				return true;
+			}
+
+			// ADMIN_LOCAL
+			const isLocalAdmin = (model.me.functions &&
+				model.me.functions.ADMIN_LOCAL && model.me.functions.ADMIN_LOCAL.scope);
+
+			if(ticket && ticket.school_id) {
+				// if parameter "ticket" is supplied, check that current user is local administrator for the ticket's school
+				return isLocalAdmin && _.contains(model.me.functions.ADMIN_LOCAL.scope, ticket.school_id);
+			}
+			return isLocalAdmin;
+		};
+		/**
+		 * Modification en masse du statut
+		 * @param newStatus : nouveau statut
+		 */
+		$scope.updateStatus = function(newStatus){
+			model.updateTicketStatus($scope.tickets.selection(), newStatus, function() {
+				notify.info('support.comment.status.modification.successful');
+				model.tickets.sync($scope.currentPage, $scope.display.filters, $scope.display.filters.school_id, $scope.sort.expression, $scope.sort.reverse);
+			}, function (e) {
+				$scope.processingData = false;
+				// validationError(e);
```

Comment:
Commenté car en javascript et typescript on a une erreur "ValidationError is not defined" (cf voir description PR)